torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -1918,15 +1918,14 @@ void InitXlaModuleBindings(py::module m) {
         [](at::Tensor& self, const at::Tensor& source) -> at::Tensor& {
           return XLANativeFunctions::set_(self, source);
         });
-  m.def("_set_xla_custom_op_name",
-        [](const at::Tensor& input, const std::string& op_name) {
+  m.def("_set_xla_custom_op_name_prefix",
```

Comment:
rename to `_set_user_metadata` and take a torch::lazy::UserMetadata if possible.

Let's pass in everything that can be forwarded to HLO here (like filename, line num, whatever Indexes etc); so no computation happens in C++.

Reply:
Can pass in a python dict (which becomes a map here) with all needed content and do creation of CustomOpNameMetadata if you feel exposing CustomOpNameMetadata through pybind11 is too much work.